pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -415,6 +454,60 @@ def queue_view(self):
         """
         return self.view_queues()
 
+    @property
+    def executor(self) -> Union[Executor, None]:
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Returns:
+            concurrent.futures.Executor:
+        """
+        if not self.run_mode.executor and self._executor is not None:
+            self._executor = None
+        return self._executor
+
+    @executor.setter
+    def executor(self, exe: Union[Executor, None]):
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Args:
+            exe (concurrent.futures.Executor):
+        """
+        if isinstance(exe, Executor):
+            self.run_mode.executor = True
+        elif exe is None:
+            self.run_mode.modal = True
+        else:
+            raise TypeError(
+                "The executor has to be derived from the concurrent.futures.Executor class."
+            )
+        self._executor = exe
+
+    @property
+    def future(self):
+        """
+        Python concurrent.futures.Future object to track the status of the execution of the job this server object is
+        attached to. This is an internal pyiron feature and most users never have to interact with the future object
+        directly.
+
+        Returns:
+             concurrent.futures.Future: future object to track the status of the execution
+        """
+        return self._future
+
+    @future.setter
+    def future(self, future_obj: Future):
```

Comment:
Leaving it orphaned might make the difficult for the code in `Project.{remove_job,wait_for_job}` to reliably check what it should it do.  Should be set to `None` therefor I think.  

Mid-term I agree there's no point in changing the run mode on a finished/running job.

Reply:
> Should be set to None therefor I think.

This is what I lean towards

Reply:
Even the example looks a the future post-facto, so I guess we can't kill it as soon as it's done. I have a vague sense of unease about this variable hanging around and causing trouble other places where assumptions are made about its existence/status, but since there are no concrete counter examples of how to break things (barring doing something really stupid like manually assigning some other random future to the attribute, but w/e, we don't need to police that), and since I don't have any alternative suggestion, I'm happy to just make a mental note of the potential pain point and move on.